src/main/java/services/car/CarComfortService.java

```diff
@@ -1,44 +1,98 @@
 package services.car;
 
-import lombok.RequiredArgsConstructor;
+import connection.Transactionable;
+import exception.CarRentalException;
+import lombok.AllArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
+import models.cars.Car;
 import models.cars.CarComfort;
+import org.apache.commons.lang3.StringUtils;
 import repository.CarComfortRepository;
-import repository.TransactionManager;
 import services.CrudGenericInterface;
 
 import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
 import java.util.UUID;
 
 @Slf4j
-@RequiredArgsConstructor
-public class CarComfortService implements CrudGenericInterface<CarComfort>, TransactionManager {
+@AllArgsConstructor
+public class CarComfortService implements CrudGenericInterface<CarComfort>, Transactionable {
     private CarComfortRepository carComfortRepository;
 
     @Override
-    public void insert(CarComfort carComfort) {
-        /*wrapIntoTransaction(() -> {
-            carComfortRepository.insert(null);
-            carComfortRepository.getAll();
-            carComfortRepository.delete(null);
-        });*/
+    public CarComfort insert(CarComfort carComfort) {
+        log.info("Trying to insert new car comfort");
+        validateCarComfort(carComfort);
+        CarComfort c = wrapIntoTransaction(() -> carComfortRepository.insert(carComfort));
+        log.info("Car comfort has been inserted successfully");
+        return c;
     }
 
     @Override
     public List<CarComfort> getAll() {
-        return null;
+        log.info("Trying to get all car comforts");
+        return carComfortRepository.getAll();
     }
 
     @Override
     public CarComfort getById(UUID id) {
-        return null;
+        log.info("Trying to get car comfort by id");
+        if (Objects.isNull(id)) {
+            log.error("Can not find car comfort. Car comfort id must be not null");
+            throw new CarRentalException("Car comfort id must be NOT null");
+        }
+        try {
+            return carComfortRepository.getById(id);
+        } catch (CarRentalException exception) {
+            log.warn("Car comfort with id {} not found", id);
+            throw new CarRentalException("Car comfort by SUCH an id must be NOT null");
```

Comment:
Could you please remove try-catch here. We will catch the exception in controller